src/persistence/db/nostr_event_offset.rs

```diff
@@ -0,0 +1,144 @@
+use super::Result;
+use crate::util::date::{self, DateTimeUtc};
+use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+use surrealdb::{engine::any::Any, Surreal};
+
+use crate::persistence::{NostrEventOffset, NostrEventOffsetStoreApi};
+
+#[derive(Clone)]
+pub struct SurrealNostrEventOffsetStore {
+    db: Surreal<Any>,
+}
+
+impl SurrealNostrEventOffsetStore {
+    const TABLE: &'static str = "nostr_event_offset";
+
+    #[allow(dead_code)]
+    pub fn new(db: Surreal<Any>) -> Self {
+        Self { db }
+    }
+}
+
+#[async_trait]
+impl NostrEventOffsetStoreApi for SurrealNostrEventOffsetStore {
+    async fn current_offset(&self) -> Result<u64> {
+        let result: Vec<NostrEventOffsetDb> = self
+            .db
+            .query("SELECT * FROM type::table($table) ORDER BY time DESC LIMIT 1")
+            .bind(("table", Self::TABLE))
+            .await?
+            .take(0)?;
+        let value = result
+            .first()
+            .map(|c| c.time.timestamp())
+            .unwrap_or(0)
```

Comment:
If I read this correctly, this means that if there is no offset in the DB, it just returns 0, which means any new offset will be higher, so this is more or less ignored in the logic.

That makes sense from my perspective, but maybe a short doc string comment mentioning it could be useful, since one could also expect to get an `Err()` in that case and that the fact that it's `0` needs to be considered when using this.